consumers/best-practices.md

```diff
@@ -41,6 +41,33 @@ Querying the current price will fail if too much time has elapsed since the last
 The SDKs expose this failure condition in an idiomatic way: for example, the Rust SDK may return `None`, and our Solidity SDK may revert the transaction.
 The SDK provides a sane default for the staleness threshold, but users may configure it to suit their use case.
 
+# Latency
+
+Developers integrating Pyth Network price feeds should account for the difference in latency between on-chain oracles and off-chain centralized exchanges.
+Although Pyth Network is designed with low latency in mind, no on-chain oracle can match the latency of an off-chain source due to the added overhead for consensus and security.
+The threat model for integrating protocols should assume that adversaries see price changes a short time before the protocol does.
+In this threat model, protocol designers should avoid situations where a Pyth price update must race against an adversary's transaction.
+The adversary has a head start in these races.
+Furthermore, sophisticated adversaries can highly optimize their network latencies, or pay miners for priority blockspace, such that they are virtually guaranteed to win these races.
+
+This situation is analogous to the one faced by market makers in the traditional finance.
+These market makers place resting orders on exchanges with the hope of earning the bid/ask spread.
+When the “true price” moves, these market makers get picked off by adverse “smart flow” that is faster than they are.
+The smart flow is balanced by two-way flow, that is, people wanting to trade for other reasons besides a price change.
+
+This analogy suggests two simple solutions to races:
+1. Configure protocol parameters to balance the losses from smart flow against the gains from two-way flow.
+   Market makers in traditional finance implement this approach by offering a bid/ask spread and limited liquidity.
+   The limited liquidity caps the losses to smart flow, while still earning profits from the two-way flow.
+   A successful market maker tunes the spread and offered liquidity to limit adverse selection from smart traders while still interacting with two-way flow.
+2. Give the protocol a "last look" to decide which transactions to accept.
+   In traditional finance, some exchanges give market makers a chance to walk back a trade offer after a someone else has requested it.
+   Protocols can implement this technique by splitting transactions into two parts: a request and a fulfillment.
+   First, the user requests an action to be performed with the current Pyth price.
+   Then, the protocol chooses to fulfil or deny that request with a time delay, at which point it can observe any changes to the Pyth price.
```

Comment:
I found this one a bit difficult to understand. Maybe worth mentioning decision executing with the new price.